helpers/calendar/ratios_and_utilization.py

```diff
@@ -159,27 +159,41 @@ def percentage_formater(self, portion, total):
         except ZeroDivisionError:
             return 0
 
-    def get_bookings_analytics_count(self, query, start, end):
+    def get_bookings_analytics_count(self, query, start, end, room_id=None):
         results = []
-        start_date, day_after_end_date = CommonAnalytics.convert_dates(self, start, end)  # noqa E501
-        start_dt = dateutil.parser.parse(start_date)
-        end_dt = dateutil.parser.parse(day_after_end_date)
-        number_of_days = (end_dt - start_dt).days
-
+        start_date, day_after_end_date = CommonAnalytics.convert_dates(
+            self, start, end)
+        formated_start_date = dateutil.parser.parse(start_date)
+        formated_end_date = dateutil.parser.parse(day_after_end_date)
+        number_of_days = (formated_end_date - formated_start_date).days
+
+        # assign exact_room a value of None in case a query is made
+        # without passing a room_id argument
+        exact_room = RoomModel.query.filter_by(
+            id=room_id).first() if room_id else None
         if number_of_days <= 30:
-            dates = CommonAnalytics.get_list_of_dates(start, number_of_days)  # noqa E501
+            room_name = exact_room.name if exact_room else None
```

Comment:
If you check if there is `exact_room` before assigning a value to a variable `room_name`, your code would have a more logical flow. Same to line 172 above.

Reply:
@Bonifase this has been worked on.